File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
     vector<int> ans(nums.size(), 0);
     int prod = 1, zc = 0; 
     for(auto i : nums){
        if(i == 0)
        zc++;
        else 
        prod *= i;
     }
     if(zc >1 )
     return ans;
     for(int i = 0; i< nums.size() ;i++){
      if(nums[i] == 0)
      ans[i] = prod;
      else if(zc == 1)
      ans[i] = 0;
      else 
      ans[i] = prod/nums[i];
     }
     return ans;
    }
};

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.java
class Solution {
    public record Point(int[] point, int distance){};
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<Point> pq = new PriorityQueue<>((x,y)-> y.distance - x.distance);
        for(int[] i :  points){
            int distance = i[0]*i[0] + i[1]*i[1];
           if(pq.size() < k) {
           pq.offer(new Point(i, distance));
           }else {
            if(distance < pq.peek().distance){
                pq.poll();
                pq.offer(new Point(i, distance));
            }
           }
        }
        int[][] ans = new int[k][2];
        for(int i = 0 ; i < k ; i++){
        ans[i] = pq.poll().point;
        }

        return ans;

    }
}
